app/services/sandbox.service.ts

```diff
@@ -17,14 +17,12 @@ export class Script {
     // and type engine in general. This forces an explicit case when you
     // use run in context, but allows type inference to mostly keep 
     // flowing
-    runInContext(context: Context): Promise<NativeResult> {
-        return new Promise((resolve, reject) => {
-            try {
-                resolve(this.nodeScript.runInContext(context));
-            } catch (err) {
-                reject(err);
-            }
-        });
+    runInContext(context: Context | Promise<Context>): Promise<NativeResult> {
+        if (!(context instanceof Promise)) {
+            context = Promise.resolve(context);
```

Comment:
I feel like it would be cleaner to create a lambda above that handles a context. If it is a promise, pass that lambda into then. If it is not a promise, just run it directly since runInContext returns a promise.